hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/UserGroupInformation.java

```diff
@@ -808,6 +810,26 @@ static void loginUserFromSubject(Subject subject) throws IOException {
       }
       loginUser = proxyUser == null ? realUser : createProxyUser(proxyUser, realUser);
 
+      String tokenFileLocation = System.getProperty(HADOOP_TOKEN_FILES);
```

Comment:
Should the logic not be to first read the conf and if null then default to the system.properties?